Backpack/src/main/java/net/skyscanner/backpack/calendar/BpkCalendar.kt

```diff
@@ -27,11 +27,17 @@ open class BpkCalendar @JvmOverloads constructor(
     calendar_view.controller = controller
     controller.updateContentCallback = calendar_view
     calendar_view.listener = this
+
+    updateYearPill(controller.startDate.year)
   }
 
   override fun onYearChanged(year: Int) {
+    this.post { updateYearPill(year) }
+  }
+
+  private fun updateYearPill(year: Int) {
```

Comment:
Why is this delegated to a new thread? Does not seem to be heavy logic and it does update UI

Reply:
Same reason as above, the logic to set the text was running on a new thread already, because it wasn't working otherwise, I just moved the whole thing to a thread (as it doesn't work otherwise in RN). I suspect that because this is called inside a scroll event we need to wait for it to finish to show the pill, but I'm not sure, might be something completely different.